File: Python/02-Buildings-of-New-York-City.py
def secondLargest(arr): # Function to find the second largest
    if len(arr) <= 1:
        return -1
    largest, secondL = arr[0], -1
    for i in range(1,len(arr)):
        if arr[i] > largest:
            secondL = largest
            largest = arr[i]
        elif arr[i] > secondL and arr[i] != largest:
            secondL = arr[i]
    return secondL
    
n = int(input()) # Taking Input
arr = list(map(int,input().split())) # Taking Input
no = secondLargest(arr) # Storing the second largest
if no == -1: # If there was no second largest number in the array
    print(-1)
else: 
    print(*[i+1 for i in range(len(arr)) if arr[i] == no]) # Print the result if there exist a second largest

File: C++/02-Buildings-of-New-York-City.cpp
#include <iostream>

using namespace std;
#include<stdio.h>
int main(){
    int n;
    cin>>n; // Taking the number of buildings
    int a[n],i,secondL,larg;
    for(i=0;i<n;i++)
        cin>>a[i]; // Taking Input
    larg =a[0];
    secondL = -1;
    for(i=1;i<n;i++){ // Finding the Second Largest
        if(a[i]>larg){
            secondL= larg;
            larg = a[i];
        }
        else if(a[i]>secondL && a[i] != larg)
            secondL = a[i];
    }
    if(secondL == -1) // If there was no second largest number in the array
        cout<<"-1";
    else{
        for(i=0;i<n;i++){
            if(a[i]==secondL){
                cout<<i+1<<" "; // Print the result if there exist a second largest
            }
        }
    }
    return 0;
}
